game-server/data/handlers/ai/instance/empyreanCrucible/RM1337AI.java

```diff
@@ -1,30 +1,30 @@
 package ai.instance.empyreanCrucible;
 
-import java.util.ArrayList;
-import java.util.List;
 import java.util.concurrent.Future;
 import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.stream.IntStream;
 
 import com.aionemu.commons.utils.Rnd;
 import com.aionemu.gameserver.ai.AIName;
+import com.aionemu.gameserver.ai.HpPhases;
 import com.aionemu.gameserver.model.gameobjects.Creature;
 import com.aionemu.gameserver.model.gameobjects.Npc;
-import com.aionemu.gameserver.model.gameobjects.player.Player;
-import com.aionemu.gameserver.skillengine.SkillEngine;
+import com.aionemu.gameserver.model.templates.npcskill.NpcSkillTargetAttribute;
 import com.aionemu.gameserver.utils.PacketSendUtility;
-import com.aionemu.gameserver.utils.PositionUtil;
 import com.aionemu.gameserver.utils.ThreadPoolManager;
 
 import ai.AggressiveNpcAI;
 
 /**
- * @author Luzien
+ * @author Luzien, w4terbomb
  */
 @AIName("rm_1337")
-public class RM1337AI extends AggressiveNpcAI {
+public class RM1337AI extends AggressiveNpcAI implements HpPhases.PhaseHandler {
 
-	private AtomicBoolean isHome = new AtomicBoolean(true);
-	private AtomicBoolean isEventStarted = new AtomicBoolean(false);
+	private final HpPhases hpPhases = new HpPhases(15, 35, 55, 75, 95);
+
+	private final AtomicBoolean isHome = new AtomicBoolean(true);
+	private final AtomicBoolean isEventStarted = new AtomicBoolean(false);
 	private Future<?> task1, task2;
```

Comment:
Maybe `private final List<Future<?>> skillTasks = new CopyOnWriteArrayList<>();` would be better? Otherwise we need to find meaningful names for both variables.